### ISSUE TITLE
Cross-referencing equations in Markdown doesn't work.

### ISSUE THREAD

**Issue by jcsutherland** [jupyterlab/jupyterlab#4039]
One of the extensions I love in jupyter is the latex extension that provides `align` constructs as well as labeling (via `\label{}`) and cross-referencing (via `\ref{}`).

# Example:
Enter this text in a markdown cell:
```
\begin{align}
    g &= \int_a^b f(x)dx \label{eq1} \\
    a &= b + c \label{eq2}
\end{align}

See (\ref{eq1})
```

* In _Jupyter notebook_ with the LaTeX nbextensions active, the equations are numbered and the cross reference shows up as a live link with the appropriate equation number.

* In _JupyterLab_ (version 0.31.8), no equation numbers are rendered, and the cross-reference is broken, showing (???).

**Comment by eseglem** [comment]
Found this issue because I was trying to do the same thing. And wasn't sure what was happening.

I figured out that within a single cell the following will work if you manually `\tag{}` the equations with a number:
```
\begin{align}
    g &= \int_a^b f(x)dx \label{eq1}\tag{1} \\
    a &= b + c \label{eq2}\tag{2}
\end{align}

See (\ref{eq1}) and (\ref{eq2})
```
But, if you `\ref{eq1}` from another cell, it will not work. It ends up the `(???)`

However, you can manually link to the equation `#mjx-eqn-<label>`, and it will work. 
```
See (\ref{eq1}) and (\ref{eq2}) does not work from another cell.
See ([1](#mjx-eqn-eq1)) and ([2](#mjx-eqn-eq2)) does work from another cell.
```

Tedious, since all the numbers are by hand, but it does work. Probably better to stick with nbextensions for automatic numbering though.

**Comment by jlperla** [comment]
Why not just turn on automatic numbering with mathjax?  See http://docs.mathjax.org/en/latest/tex.html#automatic-equation-numbering
```
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
``
Seems like it could be added to https://github.com/jupyterlab/jupyterlab/blob/8c69a974bc9797dfa61e0df7b2393790a9f1fde2/packages/mathjax2/src/index.ts#L76 and it would be a benefit to all.

I also am not sure about why https://github.com/jupyterlab/jupyterlab/blob/8c69a974bc9797dfa61e0df7b2393790a9f1fde2/packages/mathjax2/src/index.ts#L45 would be there without the autonumbering feature?

**Comment by jasongrout** [comment]
Things get trickier when we have multiple views of the same notebook on the screen at once side by side. However, I think it's worth a try to see if there is a way to scope and disambiguate references. If anyone wants to work on this, let us know and we can help you get started.

**Comment by jlperla** [comment]
Thanks @jasongrout , @arnavs may work on it if you are able to give some guidance.

A question for you:  should there be a setting on a per-notebook basis for this, or is it something you think should be always on?  i.e. what about having it in the notebook metadata JSCON
```json
"metadata": {
  "mimetype": "text/x-python",
  "OTHER": "STUFF",
  "TeX": {
      "equationNumbers": { "autoNumber" : "AMS" } }
}
```
Where that section directly maps to the appropriate mathjax?

**Comment by makslevental** [comment]
sorry i'm not clear on how this works? i have auto number on but i can't ref from other cells even after running auto-number?

**Comment by GeoMathC** [comment]
It's really boring. However, I found one markdown system by zybuluo.com, almost perfectly implemented  crossing-ref equations and system of equations. I hope this problem is fixed ASAP by Juypter Notebook.

**Comment by gideonsimpson** [comment]
I am also interested in automatic equation numbering, but I am a bit unsure from this discussion:

1. What external package I should install to do it, if that's what is called for
2. What configuration file to edit and where, if that's a better option

Using `\tag` works fine for me for manual numbering, but that's not really what I want to do.

**Comment by kurtsansom** [comment]
@gideonsimpson how does `\tag` work

**Comment by ickc** [comment]
Cf. https://github.com/ipython/ipython/issues/4113

**Comment by jcsutherland** [comment]
Just following up here.  It looks like equation numbering is now broken in Jupyter notebook as well - maybe for the same reasons?

Any plans to get this sorted out?

**Comment by pierre-haessig** [comment]
There is a Stackoverflow thread [Equation numbering in Jupyter notebooks](https://stackoverflow.com/questions/41241984/equation-numbering-in-jupyter-notebooks) where @jlperla suggestion is proposed to be run a code cell with a `%%javascript` cell magic.

However, with the switch to Mathjax 3 in Jupyterlab 4.0+, this doesn't work (see my fresh answer to this thread: https://stackoverflow.com/a/79463720/2822346). Is there a similar workaround? I've tried to new Mathjax 3 config [Automatic Equation Numbering](https://docs.mathjax.org/en/latest/input/tex/eqnumbers.html) in a code+`%%javascript`  cell (test with Jupyterlab 4.2.5), but it doesn't seem to work:

```javascript
window.MathJax = {
  tex: {
    tags: 'ams'
  }
};
```

**Comment by jcsutherland** [comment]
I'm bumping this thread in the hope that the devs will bring back this beloved feature that we lost in the updates to Jupyter Notebook.  _Please_ bring support for this back!

Any updates from anyone in the know on the likelihood of this feature coming back?

BTW, I noticed [this documentation](https://jupyterbook.org/en/stable/content/math.html) page, but can't get that to work for me.

**Comment by esarradj** [comment]
As the author of some lecture notes using Jupyter Notebook I was stuck with nbclassic for the last years because of all the features that were removed, most notably equation numbering and cross-referencing. 

Recently I came across some **solution** that works with jupterlab. It is is not a 100% drop-in replacement, but comes close enough to the features that we have nbclassic. You just need to install [jupyterlab-myst](https://github.com/jupyter-book/jupyterlab-myst) alongside jupyterlab and then you have access to [autonumbering and cross-references](https://jupyterbook.org/v1/content/math.html). Note however that the syntax is slightly different, `\eqref(my_equation)` needs to be replaced by `[](#my_equation)` and references to lines in a multiline equation will not work.

Strangely I found multiple discussions concerning how to get equation autonumbering in jupyterlab and this was NEVER even mentioned.
